kop meet ConcurrentModificationException

**Describe the bug**
If pulsar enable protocol handler kop and broker entry metadata, when start broker my meet exception, 
``` 
java.util.concurrent.CompletionException: java.util.ConcurrentModificationException
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:273) ~[?:1.8.0_201]
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:280) ~[?:1.8.0_201]
	at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:659) ~[?:1.8.0_201]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:632) ~[?:1.8.0_201]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:474) ~[?:1.8.0_201]
	at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:1962) ~[?:1.8.0_201]
	at org.apache.pulsar.broker.service.BrokerService.lambda$getManagedLedgerConfig$40(BrokerService.java:1308) ~[pulsar-broker-2.8.0-rc-202012272229.jar:2.8.0-rc-202012272229]
	at org.apache.bookkeeper.mledger.util.SafeRun$2.safeRun(SafeRun.java:49) [managed-ledger-2.8.0-rc-202012272229.jar:2.8.0-rc-202012272229]
	at org.apache.bookkeeper.common.util.SafeRunnable.run(SafeRunnable.java:36) [bookkeeper-common-4.12.0.jar:4.12.0]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_201]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_201]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-common-4.1.51.Final.jar:4.1.51.Final]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_201]
Caused by: java.util.ConcurrentModificationException
	at java.util.HashMap$HashIterator.nextNode(HashMap.java:1445) ~[?:1.8.0_201]
	at java.util.HashMap$KeyIterator.next(HashMap.java:1469) ~[?:1.8.0_201]
	at org.apache.pulsar.broker.service.BrokerService.lambda$createPersistentTopic$32(BrokerService.java:1101) ~[pulsar-broker-2.8.0-rc-202012272229.jar:2.8.0-rc-202012272229]
	at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:656) ~[?:1.8.0_201]
	... 10 more
```

**To Reproduce**
Steps to reproduce the behavior:
1. enable kop for pulsar broker
2. enable broker entey metadata for pulsar borker
3. start broker and may see this error

--- a/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java
+++ b/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java
@@ -1105,18 +1105,19 @@ public class BrokerService implements Closeable, ZooKeeperCacheListener<Policies
         }
 
         getManagedLedgerConfig(topicName).thenAccept(managedLedgerConfig -> {
+
             if (isBrokerEntryMetadataEnabled()) {
                 // init managedLedger interceptor
+                Set<BrokerEntryMetadataInterceptor> interceptors = new HashSet<>();
                 for (BrokerEntryMetadataInterceptor interceptor : brokerEntryMetadataInterceptors) {
+                    // add individual AppendOffsetMetadataInterceptor for each topic
                     if (interceptor instanceof AppendIndexMetadataInterceptor) {
-                        // add individual AppendOffsetMetadataInterceptor for each topic
-                        brokerEntryMetadataInterceptors.remove(interceptor);
-                        brokerEntryMetadataInterceptors.add(new AppendIndexMetadataInterceptor());
+                        interceptors.add(new AppendIndexMetadataInterceptor());
+                    } else {
+                        interceptors.add(interceptor);
                     }
                 }
-                ManagedLedgerInterceptor mlInterceptor =
-                        new ManagedLedgerInterceptorImpl(brokerEntryMetadataInterceptors);
-                managedLedgerConfig.setManagedLedgerInterceptor(mlInterceptor);
+                managedLedgerConfig.setManagedLedgerInterceptor(new ManagedLedgerInterceptorImpl(interceptors));
             }
 
             managedLedgerConfig.setCreateIfMissing(createIfMissing);

--- a/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java
+++ b/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java
@@ -83,7 +83,6 @@ import org.apache.bookkeeper.mledger.ManagedLedgerConfig;
 import org.apache.bookkeeper.mledger.ManagedLedgerException;
 import org.apache.bookkeeper.mledger.ManagedLedgerException.ManagedLedgerNotFoundException;
 import org.apache.bookkeeper.mledger.ManagedLedgerFactory;
-import org.apache.bookkeeper.mledger.intercept.ManagedLedgerInterceptor;
 import org.apache.bookkeeper.mledger.util.Futures;
 import org.apache.bookkeeper.util.ZkUtils;
 import org.apache.commons.lang3.StringUtils;

